[14.0] Improve computation of sepa boolean, update move generation, improve views

alexis-via opened the pull request:
Improve computation of sepa on account.payment.order: check IBAN is in SEPA zone
Update move line generation to get transfer account from bank journal
Update payment mode configuration accordingly (3 fields removed)
Several improvements in payment order tree and form view

alexis-via committed:
Improve computation of sepa on account.payment.order: check IBAN is in SEPA zone

Update move line generation to get transfer account from bank journal
Update payment mode configuration accordingly (3 fields removed)
Several improvements in payment order tree and form view

pedrobaeza reviewed (commented):
Why removing the offsetting account + journal? I don't think the journal selected for the payment order serves for the same, as for SEPA orders, you select the bank journal, but you want to register the intermediate move on other journal + account

alexis-via committed:
account_payment_order: remove done state (migration script provided)

Don't set reference_type field on account.move required=True, because it
causes a lot of problems in other modules that don't depend on this one.
Setting it as required in the view is enough.

alexis-via committed:
account_payment_partner: add optional="hide" on payment mode in invoice tree view

alexis-via committed:
FIX crash when communication=null on payment line

alexis-via committed:
account_payment_order: payment_mode_id readonly when state != draft

alexis-via committed:
account_payment_mode: allow cash journals

alexis-via commented:
@pedrobaeza Well, it's the philosophy of odoo v14!
- The offsetting account is the payment_debit_account_id/payment_credit_account_id of the bank journal
- the offsetting move is also the same bank journal.
This is what happens when you click on "Register Payment" on an invoice. I can understand that few people may prefer to have a different journal for the offsetting move... but most people will be OK with this I think.

pedrobaeza commented:
Please check Travis and reduce the number of commits.

alexis-via committed:
[FIX] account_banking_sepa_direct_debit: fix crash when printing a mandate with empty partner

NL66278 commented on account_banking_pain_base/views/account_payment_order.xml:
```diff
@@ -21,5 +21,19 @@
                 />
             </field>
         </field>
-    </record>
```
I think this shifting of the closing `</record>` is a mistake

alexis-via replied:
As you know, the formatting is done by black/isort and all those automated tools. So, if you're not happy with XML reformatting, you can report this upstream :)

NL66278 replied:
Strange, if I change the formatting to what I would expect, it is perfectly fine with prettier etc. Is not blocking of course, but it just struck me as odd.

NL66278 commented on account_banking_pain_base/views/account_payment_order.xml:
```diff
@@ -21,5 +21,19 @@
+            ref="account_payment_order.account_payment_order_tree"
+        />
+        <field name="arch" type="xml">
+                <field name="description" position="after">
```
And here indenta are wrong. DId you use tabs instead of spaces?

alexis-via replied:
same answer

NL66278 commented on account_banking_pain_base/views/bank_payment_line_view.xml:
```diff
@@ -17,4 +17,15 @@
+        <field name="inherit_id" ref="account_payment_order.bank_payment_line_tree" />
+        <field name="arch" type="xml">
+            <field name="communication" position="after">
+                    <field name="priority" optional="hide" />
```
Same here,

alexis-via replied:
same answer

NL66278 reviewed (approved):
:+1: Functionally seems good to me. Some remarks on xml formatting. And some of our customers are missing the 'done' state that was set in the pas when payments were finally reconciled.

NL66278 commented:
@alexis-via What is the work still being done on this?

alexis-via commented:
@NL66278 This is the branch I deploy for my v14 customers. There is no special work being done here at the moment, but this branch consolidate all the work I made so far to fix and improve bank-payment for v14.

alexis-via commented:
BTW, one thing left to do it to fix Travis. I'll try to find time to work on that in the coming days, to merge this PR.

alexis-via committed:
Update test suite

Fix bug in account_banking_mandate when a res.partner.bank is not attached to a particular company (company_id is NOT required on res.partner.bank)

alexis-via commented:
Tests are ok on my laptop... so I hope Travis will be green.

alexis-via commented:
Travis is green, so I propose to merge this.

alexis-via commented:
/ocabot merge nobump

ProcessControl-MTG commented:
Aunque la funcionalidad es similar, no sirve para lo mismo... La cuenta de compensación en muchos casos se utilizaba para las remesas al descuento contra la 431. Odoo no permite indicar esta cuenta en el diario de banco cómo cuenta transitoria...Por lo que el balance sería incorrecto. @alexis-via

rousseldenis commented:
> @pedrobaeza Well, it's the philosophy of odoo v14!
> 
> * The offsetting account is the payment_debit_account_id/payment_credit_account_id of the bank journal
> * the offsetting move is also the same bank journal.
>   This is what happens when you click on "Register Payment" on an invoice. I can understand that few people may prefer to have a different journal for the offsetting move... but most people will be OK with this I think.

@alexis-via We have a customer that requires a different journal than the bank one.

Could we revert this or find a way to keep both possible ?
